src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts

```diff
@@ -1,99 +1,304 @@
-import { Component, EventEmitter, Inject, OnDestroy, OnInit } from '@angular/core';
-import { MatDialogRef } from '@angular/material/dialog';
-import { CredentialsService } from '../../../core/services/credentials/credentials.service';
-import { DOCUMENT } from '@angular/common';
+import { Component, HostBinding, Inject, OnDestroy, OnInit } from '@angular/core';
 import { AutoSubscription, AutoSubscriptions } from 'auto-subscriptions';
-import { tap, filter, timeout } from 'rxjs/operators';
-import { combineLatest } from 'rxjs';
-import { saveAs } from 'file-saver';
-import { IToolsConfig, toolsConfig } from '../models/tools-config';
-import { Store } from '@ngrx/store';
-import { selectIsMinimalistViewMode, SetMinimalistViewModeAction } from '@ansyn/map-facade';
+import { select, Store } from '@ngrx/store';
+import {
+	floationMenuClassNameForExport,
+	imageryStatusClassNameForExport, selectActiveMapId,
+	selectIsMinimalistViewMode,
+	SetMinimalistViewModeAction, SetToastMessageAction
+} from '@ansyn/map-facade';
 import { LoggerService } from '../../../core/services/logger.service';
+import { debounceTime, filter, tap, map, mergeMap, catchError, finalize } from 'rxjs/operators';
+import { saveAs } from 'file-saver';
 import { toBlob } from 'dom-to-image';
+import { MatDialogRef } from '@angular/material/dialog';
+import { DOCUMENT } from '@angular/common';
+import { IToolsConfig, toolsConfig } from '../models/tools-config';
+import { annotationsClassNameForExport } from '@ansyn/ol';
+import { measuresClassNameForExport } from '../../../plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer';
+import { IExportMapData, IExportMapMetadata, ImageryCommunicatorService, toDegrees } from '@ansyn/imagery';
+import { jsPDF } from "jspdf";
+import { TranslateService } from '@ngx-translate/core';
+import { IOverlay } from '../../../overlays/models/overlay.model';
+import { Observable, of, EMPTY } from 'rxjs';
 
+enum GraphicExportEnum {
```

Comment:
i think moving this to a seperate config file would be more right

Reply:
i don't agree. it is not relevant to any file except this one.

Reply:
so don't you think it should have config of its own?
it makes the file less overwhelming